src/main/java/com/sounganization/botanify/domain/chat/repository/ChatMessageCustomRepositoryImpl.java

```diff
@@ -76,4 +76,43 @@ public Page<ChatMessage> findMessagesByRoomIdWithPagination(Long roomId, Pageabl
 
         return new PageImpl<>(messages, pageable, total);
     }
+
+    @Override
+    public int softDeleteMessagesOlderThan(LocalDateTime cutoffDate, int batchSize) {
+        QChatMessage message = QChatMessage.chatMessage;
+
+        List<Long> messageIds = jpaQueryFactory
+                .select(message.id)
+                .from(message)
+                .where(message.createdAt.before(cutoffDate)
+                        .and(message.deletedYn.isFalse()))
+                .limit(batchSize)
+                .fetch();
+
+        if (messageIds.isEmpty()) {
+            return 0;
+        }
+
+        return (int) jpaQueryFactory
```

Comment:
what's the exact purpose typecasting to 'int'?

Reply:
The reason is that QueryDSL's "execute()" method returns a long and our method signature "softDeleteMessagesOlderThan()" requires an int return type. For our use case, casting is safe as we're using batch size.